File: backend/core/src/modules/document-comments/document-comments.controller.ts
import { Body, Controller, Delete, Get, Param, Patch, Post, Query, Req, UseGuards } from '@nestjs/common';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';
import { DocumentCommentsService } from './document-comments.service';

@UseGuards(JwtAuthGuard)
@Controller('document-comments')
export class DocumentCommentsController {
  constructor(private readonly documentCommentsService: DocumentCommentsService) {}

  @Get()
  findAll(@Req() req: any, @Query('documentId') documentId?: string) {
    if (!documentId) return [];
    return this.documentCommentsService.findAllForUser(req.user?.userId, documentId);
  }

  @Get(':id')
  findOne(@Req() req: any, @Param('id') id: string) {
    return this.documentCommentsService.findOneForUser(req.user?.userId, id);
  }

  @Post()
  create(@Req() req: any, @Body() dto: any) {
    return this.documentCommentsService.createForUser(req.user?.userId, dto);
  }

  @Patch(':id')
  update(@Req() req: any, @Param('id') id: string, @Body() dto: any) {
    return this.documentCommentsService.updateForUser(req.user?.userId, id, dto);
  }

  @Delete(':id')
  async remove(@Req() req: any, @Param('id') id: string) {
    await this.documentCommentsService.removeForUser(req.user?.userId, id);
    return { success: true };
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/DocumentCommentsController.java
package store.pjcloud.cowrite.core.controller;

import java.util.List;
import java.util.Map;
import java.util.UUID;
import org.springframework.web.bind.annotation.*;
import store.pjcloud.cowrite.core.entity.DocumentComment;
import store.pjcloud.cowrite.core.security.SecurityUtils;
import store.pjcloud.cowrite.core.service.DocumentCommentsService;

@RestController
@RequestMapping("/document-comments")
public class DocumentCommentsController {
    private final DocumentCommentsService service;

    public DocumentCommentsController(DocumentCommentsService service) {
        this.service = service;
    }

    @GetMapping
    public List<DocumentComment> list(@RequestParam("documentId") UUID documentId) {
        UUID userId = SecurityUtils.requireUserId();
        return service.findAllForUser(userId, documentId);
    }

    @GetMapping("/{id}")
    public DocumentComment get(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        return service.findOneForUser(userId, id);
    }

    @PostMapping
    public DocumentComment create(@RequestBody DocumentComment body) {
        UUID userId = SecurityUtils.requireUserId();
        return service.createForUser(userId, body);
    }

    @PatchMapping("/{id}")
    public DocumentComment update(@PathVariable("id") UUID id, @RequestBody DocumentComment body) {
        UUID userId = SecurityUtils.requireUserId();
        return service.updateForUser(userId, id, body);
    }

    @DeleteMapping("/{id}")
    public Map<String, Object> delete(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        service.removeForUser(userId, id);
        return Map.of("success", true);
    }
}
